scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -3080,18 +3083,18 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
     trees
   }
 
-  def blockExpr(): Term = autoPos {
+  def blockExpr(isBlockOptional: Boolean = false): Term = autoPos {
```

Comment:
I found that for parsing cases we use `expr(location = BlockStat, ...)`: https://github.com/scalameta/scalameta/blob/c058f6dfd2d220a8cef2e44669f3adb393ff6d6d/scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala#L3124

I'm wondering if the same approach would work for `function` and `implicitClosure`

Reply:
I tried that and seems like `expr` usage instead of `blockExpr` fixes that problem - [commit](https://github.com/dos65/scalameta/commit/37f1fd38a7931d944d01b36dcf2098f7dabff4b3).
I'm not fully sure if it fixes any additional cases but at least it looks more generic.
@tgodzik wdyt?

Reply:
That's a good idea! Thanks! And also changed!

Reply:
Actually, it doesn't work after checking again. Reverted to the last version.